### PR TITLE
[ANCHOR-377] Switch business event delivery to infinite retry

### PR DESCRIPTION
**Pull request by JiahuiWho**
### Description

Separate client event processor from business event processor 
Implement infinite retries with exponential backoff when delivering events to the business server. 

### Testing

- `./gradlew test`

### CODE REVIEW

File: platform/src/test/kotlin/org/stellar/anchor/platform/event/CallbackApiEventProcessorTest.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  @MockK(relaxed = true) lateinit var event: AnchorEvent
+  @MockK(relaxed = true) lateinit var eventHandler: CallbackApiEventHandler
+  @MockK(relaxed = true) lateinit var eventService: EventService
+  @MockK(relaxed = true) lateinit var backoffTimer: ExponentialBackoffTimer
+  private lateinit var eventProcessor: CallbackApiEventProcessor
+
+  @BeforeEach
+  fun setup() {
+    MockKAnnotations.init(this)
+    //    eventHandler = Mockito.mock(CallbackApiEventHandler::class.java)
```
Comment (lijamie98) [inline platform/src/test/kotlin/org/stellar/anchor/platform/event/CallbackApiEventProcessorTest.kt:24]: Remove?

File: platform/src/test/kotlin/org/stellar/anchor/platform/event/CallbackApiEventProcessorTest.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  @Test
+  fun `test event handled successfully without retry`() {
+    every { eventHandler.handleEvent(event) } returns true
+    eventProcessor.handleEventWithRetry(event)
+    verify(exactly = 1) { eventHandler.handleEvent(any()) }
+    verify(exactly = 1) { eventProcessor.incrementProcessedCounter() }
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = [1, 3, 5, 10, 100])
```
Comment (lijamie98) [inline platform/src/test/kotlin/org/stellar/anchor/platform/event/CallbackApiEventProcessorTest.kt:46]: Will `100` takes very long to test?

File: platform/src/main/java/org/stellar/anchor/platform/event/EventProcessor.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    } catch (Exception ex) {
+      // This is unexpected, so we need to restart the consumer.
+      Log.errorEx(ex);
+    } finally {
+      queueSession.close();
+      infoF("Closing queue session [{}]", queueSession.getSessionName());
+    }
+  }
+
+  abstract void handleEventWithRetry(AnchorEvent event);
```
Comment (lijamie98) [inline platform/src/main/java/org/stellar/anchor/platform/event/EventProcessor.java:82]: Can we rename to `handleEvent`?

File: platform/src/main/java/org/stellar/anchor/platform/event/CallbackApiEventProcessor.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+          } catch (InterruptedException e) {
+            currentThread().interrupt();
+          }
+        }
+      } catch (IOException ioex) {
+        Log.errorEx(ioex);
+        try {
+          getBackoffTimer().backoff();
+        } catch (InterruptedException e) {
+          currentThread().interrupt();
```
Comment (lijamie98) [inline platform/src/main/java/org/stellar/anchor/platform/event/CallbackApiEventProcessor.java:45]: If the thread is already interrupted, should we `interrupt` again?

File: platform/src/main/java/org/stellar/anchor/platform/event/CallbackApiEventProcessor.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      String name,
+      EventService.EventQueue eventQueue,
+      EventService eventService,
+      CallbackApiEventHandler eventHandler) {
+    super(name, eventQueue, eventService);
+    this.eventHandler = eventHandler;
+  }
+
+  @Override
+  void handleEventWithRetry(AnchorEvent event) {
```
Comment (lijamie98) [inline platform/src/main/java/org/stellar/anchor/platform/event/CallbackApiEventProcessor.java:26]: Can we add some comments here for how the retry is handled?

File: platform/src/test/kotlin/org/stellar/anchor/platform/event/CallbackApiEventProcessorTest.kt (hunk 2 above)
Comment (JiahuiWho) [reply platform/src/test/kotlin/org/stellar/anchor/platform/event/CallbackApiEventProcessorTest.kt:46]: No .. it pretty fast

File: platform/src/main/java/org/stellar/anchor/platform/event/EventProcessor.java (hunk 3 above)
Comment (JiahuiWho) [reply platform/src/main/java/org/stellar/anchor/platform/event/EventProcessor.java:82]: I want to differentiate this from the `handleEvent()` in `eventHandler`, where the inheritors should specifically hand retry here

File: platform/src/main/java/org/stellar/anchor/platform/event/CallbackApiEventProcessor.java (hunk 4 above)
Comment (JiahuiWho) [reply platform/src/main/java/org/stellar/anchor/platform/event/CallbackApiEventProcessor.java:45]: It's a common approach. This sets the interrupt flag of the thread, so higher-level interrupt handlers will notice it and can handle it appropriately.

**Pull request merged**